### DISCUSSION TITLE
Using NetTopologySuite with EF Core 5 on Linux ARM

### DISCUSSION THREAD

**Discussion by d79ima** [NetTopologySuite/NetTopologySuite#537] [Q&A] [answered]
Has anyone been successful in using NetTopologySuite in a .NET Core 3.1 app running on Linux with EF Core 5 and Sqlite?
My app is experiencing a segmentation fault as soon as i try to call DbContext.Migrate()

**Comment by airbreather** [comment]
Would you be in a position to create a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) that we could use to investigate this?

CC: @bricelam

**Reply by d79ima** [reply]
i will give it a try

**Comment by d79ima** [comment]
Very simple hello world app using Visual Studio 2019 that demonstrates segmentation fault.
publish for linux-arm.
[TestNetTopologySuite.zip](https://github.com/NetTopologySuite/NetTopologySuite/files/6824799/TestNetTopologySuite.zip)
Example not using latest version of nugets, but i just also tried with the latest 5.0.8 instead of 5.0.1 and same result.

![TestNetTopologySuite_SegmentationFault](https://user-images.githubusercontent.com/35668690/125823629-e7d36291-f886-4622-a33c-273446f8e9e0.png)

Please let me know whether you can reproduce this issue.

**Reply by d79ima** [reply]
I also published this example for linux-x64 and tested on Ubuntu 18.04 LTS machine.
It threw exception that mod_spatialite.so was missing.
After i manually installed libsqlite3-mod-spatialite, the example app started working. No segmentation fault.

**Reply by airbreather** [reply]
I don't have readily available access to a `linux-arm` device to test this soon.  I will need a few days to get my RPi set back up again, but I expect that I will be able to.

> After i manually installed libsqlite3-mod-spatialite, the example app started working. No segmentation fault.

This is expected -- see https://docs.microsoft.com/en-us/ef/core/providers/sqlite/spatial for this and other important documentation about using spatial data in SQLite with EF Core.

**Reply by d79ima** [reply]
Thanks for taking a look. Looked at my RPi /usr/lib folder and libsqlite3-mod-spatialite is installed there.

**Comment by bricelam** [comment, marked as answer]
[A workaround](https://docs.microsoft.com/en-us/ef/core/providers/sqlite/spatial#installing-spatialite) is mentioned the docs:

> Unfortunately, newer versions of PROJ (a dependency of SpatiaLite) are incompatible with EF's default SQLitePCLRaw bundle. You can work around this by using the system SQLite library instead.
``` xml
<ItemGroup>
  <!-- Use bundle_sqlite3 instead with SpatiaLite on macOS and Linux -->
  <!--<PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="3.1.0" />-->
  <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite.Core" Version="3.1.0" />
  <PackageReference Include="SQLitePCLRaw.bundle_sqlite3" Version="2.0.4" />

  <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite.NetTopologySuite" Version="3.1.0" />
</ItemGroup>
```
> On macOS, you'll also need set an environment variable before running your app so it uses Homebrew's version of SQLite. In Visual Studio for Mac, you can set this under Project > Project Options > Run > Configurations > Default
```sh
DYLD_LIBRARY_PATH=/usr/local/opt/sqlite/lib
```

**Reply by d79ima** [reply]
thanks will give that one a try

**Reply by d79ima** [reply]
I tried workaround suggested by Microsoft and now getting a System.DllNotFoundException: Unable to load shared library 'sqlite3' or one of its dependencies.
![sqlite3loadException](https://user-images.githubusercontent.com/35668690/125859775-a51b39c8-83ad-434f-8a22-9bed13016cc8.png)

**Reply by d79ima** [reply]
finally got this to work. had to install libsqlite3-dev package on target device
